modules/apps/collaboration/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/service/MBDiscussionMBMessageServiceWrapper.java

```diff
@@ -0,0 +1,155 @@
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.CompanyThreadLocal;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.auth.PrincipalThreadLocal;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.service.BaseServiceImpl;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceWrapper;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sergio González
+ */
+@Component(immediate = true, service = ServiceWrapper.class)
+public class MBDiscussionMBMessageServiceWrapper
+	extends MBMessageServiceWrapper {
+
+	public MBDiscussionMBMessageServiceWrapper() {
+		super(null);
+	}
+
+	public MBDiscussionMBMessageServiceWrapper(
+		MBMessageService mbMessageService) {
+
+		super(mbMessageService);
+	}
+
+	@Override
+	public MBMessage addDiscussionMessage(
+			long groupId, String className, long classPK, long threadId,
+			long parentMessageId, String subject, String body,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		User user = _getGuestOrUser();
+
+		MBDiscussionPermission.check(
+			_getPermissionChecker(), user.getCompanyId(),
+			serviceContext.getScopeGroupId(), className, classPK,
+			ActionKeys.ADD_DISCUSSION);
+
+		return super.addDiscussionMessage(
+			groupId, className, classPK, threadId, parentMessageId, subject,
+			body, serviceContext);
+	}
+
+	@Override
+	public void deleteDiscussionMessage(long messageId) throws PortalException {
+		MBDiscussionPermission.check(
+			_getPermissionChecker(), messageId, ActionKeys.DELETE_DISCUSSION);
+
+		super.deleteDiscussionMessage(messageId);
+	}
+
+	@Override
+	public MBMessage updateDiscussionMessage(
+			String className, long classPK, long messageId, String subject,
+			String body, ServiceContext serviceContext)
+		throws PortalException {
+
+		MBDiscussionPermission.check(
+			_getPermissionChecker(), messageId, ActionKeys.UPDATE_DISCUSSION);
+
+		return super.updateDiscussionMessage(
+			className, classPK, messageId, subject, body, serviceContext);
+	}
+
+	private User _getGuestOrUser() throws PortalException {
+		try {
+			return _getUser();
+		}
+		catch (PrincipalException pe) {
+			try {
+				return _userLocalService.getDefaultUser(
+					CompanyThreadLocal.getCompanyId());
+			}
+			catch (Exception e) {
+				throw pe;
+			}
+		}
+	}
+
+	private PermissionChecker _getPermissionChecker()
```

Comment:
Hi @brianchandotcom,

These private methods will go away as soon as we extract the `MBMessage` service. We will include these changes in the new service and use the inherited version of said methods.